src/main/java/kai/util/Parser.java

```diff
@@ -0,0 +1,61 @@
+package kai.util;
+
+import kai.task.Task;
+import kai.task.Todo;
+import kai.task.Deadline;
+import kai.task.TaskList;
+import kai.task.Event;
+
+public class Parser {
+    /**
+     * Reads user input of tasks and parse it into the current TaskList.
+     *
+     * @param inputCommand User input from terminal.
+     * @param tasklist Current TaskList.
+     */
+    public static void parse(String inputCommand, TaskList tasklist) {
+        if (inputCommand.equals("list")) {
+            tasklist.listDownTask();
+        } else if (inputCommand.startsWith("mark")) {
+            char pos = inputCommand.charAt(5);
```

Comment:
What if the user tries to mark the 10th task?

Reply:
I did not foresee this potential bug. I made changes to the code. Thank you!